src/math/lp/column.h

```diff
@@ -37,17 +37,21 @@ inline std::ostream& operator<<(std::ostream& out, lconstraint_kind k) {
     return out << "??";
 }
 
-inline bool compare(const std::pair<mpq, var_index> & a, const std::pair<mpq, var_index> & b) {
+inline bool compare(const std::pair<mpq, lpvar> & a, const std::pair<mpq, lpvar> & b) {
     return a.second < b.second;
 }
-
-class ul_pair {
+class lar_term; // forward definition
+class column {
     u_dependency* m_lower_bound_witness = nullptr;
     u_dependency* m_upper_bound_witness = nullptr;
     bool          m_associated_with_row = false;  
+    lpvar         m_j; //the column index
```

Comment:
the m_columns array stores column objects. The index m_j is always going to be the same as the index where the column is stored.
Do you evern need to access m_j after having lost track of how you accessed m_columns?

Reply:
Addressed